From dd45e2be1243132f53c51e16f57b4c777636d40f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20Marczykowski-G=C3=B3recki?=
 <marmarek@invisiblethingslab.com>
Date: Wed, 28 Jun 2017 16:00:56 +0200
Subject: [PATCH] cpu: define sub-leaf 0 for leaf 7 in cpu_map.xml
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Organization: Invisible Things Lab
Cc: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>

CPUID leaf 7 is sub-leaf aware. Add missing attribute.

Signed-off-by: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>
---
 src/cpu/cpu_map.xml | 58 ++++++++++++++++++++++++++---------------------------
 1 file changed, 29 insertions(+), 29 deletions(-)

diff --git a/src/cpu/cpu_map.xml b/src/cpu/cpu_map.xml
index 7d5540a..585e934 100644
--- a/src/cpu/cpu_map.xml
+++ b/src/cpu/cpu_map.xml
@@ -195,93 +195,93 @@
 
     <!-- cpuid function 0x7 ecx 0x0 features -->
     <feature name='fsgsbase'>
-      <cpuid eax_in='0x07' ebx='0x00000001'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x00000001'/>
     </feature>
     <feature name='tsc_adjust'> <!-- tsc-adjust -->
-      <cpuid eax_in='0x07' ebx='0x00000002'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x00000002'/>
     </feature>
     <feature name='bmi1'>
-      <cpuid eax_in='0x07' ebx='0x00000008'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x00000008'/>
     </feature>
     <feature name='hle'>
-      <cpuid eax_in='0x07' ebx='0x00000010'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x00000010'/>
     </feature>
     <feature name='avx2'>
-      <cpuid eax_in='0x07' ebx='0x00000020'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x00000020'/>
     </feature>
     <feature name='smep'>
-      <cpuid eax_in='0x07' ebx='0x00000080'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x00000080'/>
     </feature>
     <feature name='bmi2'>
-      <cpuid eax_in='0x07' ebx='0x00000100'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x00000100'/>
     </feature>
     <feature name='erms'>
-      <cpuid eax_in='0x07' ebx='0x00000200'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x00000200'/>
     </feature>
     <feature name='invpcid'>
-      <cpuid eax_in='0x07' ebx='0x00000400'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x00000400'/>
     </feature>
     <feature name='rtm'>
-      <cpuid eax_in='0x07' ebx='0x00000800'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x00000800'/>
     </feature>
     <feature name='cmt'>
-      <cpuid eax_in='0x07' ebx='0x00001000'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x00001000'/>
     </feature>
     <feature name='mpx'>
-      <cpuid eax_in='0x07' ebx='0x00004000'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x00004000'/>
     </feature>
     <feature name='avx512f'>
-      <cpuid eax_in='0x07' ebx='0x00010000'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x00010000'/>
     </feature>
     <feature name='avx512dq'>
-      <cpuid eax_in='0x07' ebx='0x00020000'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x00020000'/>
     </feature>
     <feature name='rdseed'>
-      <cpuid eax_in='0x07' ebx='0x00040000'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x00040000'/>
     </feature>
     <feature name='adx'>
-      <cpuid eax_in='0x07' ebx='0x00080000'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x00080000'/>
     </feature>
     <feature name='smap'>
-      <cpuid eax_in='0x07' ebx='0x00100000'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x00100000'/>
     </feature>
     <feature name='avx512ifma'>
-      <cpuid eax_in='0x07' ebx='0x00200000'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x00200000'/>
     </feature>
     <feature name='clflushopt'>
-      <cpuid eax_in='0x07' ebx='0x00800000'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x00800000'/>
     </feature>
     <feature name='avx512pf'>
-      <cpuid eax_in='0x07' ebx='0x04000000'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x04000000'/>
     </feature>
     <feature name='avx512er'>
-      <cpuid eax_in='0x07' ebx='0x08000000'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x08000000'/>
     </feature>
     <feature name='avx512cd'>
-      <cpuid eax_in='0x07' ebx='0x10000000'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x10000000'/>
     </feature>
     <feature name='avx512bw'>
-      <cpuid eax_in='0x07' ebx='0x40000000'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x40000000'/>
     </feature>
     <feature name='avx512vl'>
-      <cpuid eax_in='0x07' ebx='0x80000000'/>
+      <cpuid eax_in='0x07' ecx_in='0' ebx='0x80000000'/>
     </feature>
 
     <feature name='avx512vbmi'>
-      <cpuid eax_in='0x07' ecx='0x00000002'/>
+      <cpuid eax_in='0x07' ecx_in='0' ecx='0x00000002'/>
     </feature>
     <feature name='pku'>
-      <cpuid eax_in='0x07' ecx='0x00000008'/>
+      <cpuid eax_in='0x07' ecx_in='0' ecx='0x00000008'/>
     </feature>
     <feature name='ospke'>
-      <cpuid eax_in='0x07' ecx='0x00000010'/>
+      <cpuid eax_in='0x07' ecx_in='0' ecx='0x00000010'/>
     </feature>
 
     <feature name='avx512-4vnniw'>
-      <cpuid eax_in='0x07' edx='0x00000004'/>
+      <cpuid eax_in='0x07' ecx_in='0' edx='0x00000004'/>
     </feature>
     <feature name='avx512-4fmaps'>
-      <cpuid eax_in='0x07' edx='0x00000008'/>
+      <cpuid eax_in='0x07' ecx_in='0' edx='0x00000008'/>
     </feature>
 
     <!-- Processor Extended State Enumeration sub leaf 1 -->
-- 
2.7.4

